### PR TITLE
Limit the maximum number of requests which can issued concurrently to the chain

### PR DESCRIPTION
**Pull request by romac**
This PR targets #3341 

---

Controlled via the `max_concurrency` setting in the chain configuration.

### CODE REVIEW

**Comment by romac** [comment]
I should probably use a `CondVar` + `Mutex` here instead of pulling in `resevents`. Will get to it later.

File: crates/relayer/src/chain/handle/base.rs [hunk 1]
```diff
@@ -68,7 +72,17 @@ impl Display for BaseChainHandle {
 
 impl BaseChainHandle {
     pub fn new(chain: Arc<ChainImpl>) -> Self {
-        Self { chain }
+        // The semaphore is initialized with the maximum number of concurrent requests.
+        // If that number was specified as 0, then we use the maximum amount of concurrency,
+        // and effectively disable the limit.
```
Comment (seanchen1991) [inline crates/relayer/src/chain/handle/base.rs:77]: Should this fact be mentioned in the `config.toml` for the `max_concurrency` parameter?

Comment (romac) [reply crates/relayer/src/chain/handle/base.rs:77]: Yes, good catch!

Comment (seanchen1991) [reply crates/relayer/src/chain/handle/base.rs:77]: [920fd69](https://github.com/informalsystems/hermes/pull/3341/commits/920fd697b2d2b16d627ad1cfec1fa31a28c4b10a)

**Pull request merged**